src/Nethermind/Nethermind.Synchronization.Test/ParallelSync/MultiSyncModeSelectorTests.cs

```diff
@@ -562,12 +562,12 @@ public ScenarioBuilder WhateverTheSyncProgressIs()
                     IfThisNodeIsProcessingAlreadyDownloadedBlocksInFullSync();
                     IfThisNodeIsInTheMiddleOfFastSyncAndFastBlocks();
                     IfThisNodeFinishedFastBlocksButNotFastSync();
-                    IfThisNodeJustFinishedFastBlocksAndFastSync();
+                    IfThisNodeJustFinishedFastBlocksAndFastSync(FastBlocksState.FinishedHeaders);
                     IfThisNodeFinishedStateSyncButNotFastBlocks();
                     IfThisNodeJustFinishedStateSyncButNeedsToCatchUpToHeaders();
-                    IfThisNodeJustFinishedStateSyncAndFastBlocks();
-                    IfThisNodeJustStartedFullSyncProcessing();
-                    IfThisNodeRecentlyStartedFullSyncProcessing();
+                    IfThisNodeJustFinishedStateSyncAndFastBlocks(FastBlocksState.FinishedHeaders);
```

Comment:
I cannto see any tests for Finished Bodies -> repeat yourself here and different scenarios just to make sure that everything is covered